pkgs/development/python-modules/scan-build/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, buildPythonPackage, fetchFromGitHub, isPy27, nose }:
+
+buildPythonPackage rec {
+  pname = "scan-build";
+  version = "2.0.18";
+  disabled = isPy27;
+
+  src = fetchFromGitHub{
+    owner = "rizsotto";
+    repo = pname;
+    rev = version;
+    sha256 = "1fpmpg41314qwy6prwknh0hkclcqc1lm7lbmh4r8scsfzs3zch6b";
+  };
+
+  checkInputs = [ nose ];
+  checkPhase = "nosetests tests/unit";
```

Comment:
not sure what the nose equivalent would be, but I would do this in pytest:
```
  # ignore odd clang tests
  checkInputs  =  [ pytest clang ];
  checkPhase = ''
    pytest tests/unit -k 'not sip and not get_clang_arguments_fails'
  '';
```

Reply:
Alrighty, I've done the equivalent, tests are now passing.

Reply:
I disabled two tests, the 'sip' test is still failing:
```
builder for '/nix/store/5srpjrm27mfscprqxipgf07f64pd1rsy-python3.8-scan-build-2.0.18.drv' failed with exit code 1; last 10 log lines:
      self.assertTrue(sut.is_preload_disabled(osx))
  AssertionError: False is not true
  -------------------- >> begin captured logging << --------------------
  root: DEBUG: exec command ['csrutil', 'status'] in /build/source
  --------------------- >> end captured logging << ---------------------

  ----------------------------------------------------------------------
  Ran 74 tests in 0.676s

  FAILED (SKIP=7, failures=1)
builder for '/nix/store/1lh6pv3c46knrm0pscqzvabd8yc9pkl3-python3.7-scan-build-2.0.18.drv' failed with exit code 1; last 10 log lines:
      self.assertTrue(sut.is_preload_disabled(osx))
  AssertionError: False is not true
  -------------------- >> begin captured logging << --------------------
  root: DEBUG: exec command ['csrutil', 'status'] in /build/source
  --------------------- >> end captured logging << ---------------------

  ----------------------------------------------------------------------
  Ran 74 tests in 0.696s

  FAILED (SKIP=7, failures=1)
cannot build derivation '/nix/store/i2h96aq1cqbhffdm8pwvmrhcc8v1jmz9-env.drv': 2 dependencies couldn't be built
[1 built (2 failed), 6 copied (843.8 MiB), 138.0 MiB DL]
error: build of '/nix/store/i2h96aq1cqbhffdm8pwvmrhcc8v1jmz9-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/80374
2 package failed to build:
python37Packages.scan-build python38Packages.scan-build
```

Reply:
Huh, that's not failing for me. I'll look into it.

Reply:
not sure where scrutil is coming from